File: javascript/11_error_handling.js
// 11 error handling

function divide(a, b) {
  if (b === 0) {
    throw new Error("Cannot divide by zero");
  }
  return a / b;
}

try {
  console.log(divide(10, 2));
  console.log(divide(10, 0));
} catch (err) {
  console.log("Error:", err.message);
} finally {
  console.log("This always runs");
}

// async style later

File: python/11_error_handling.py
# 11 error handling

try:
    num = int(input("Enter a number: "))
    result = 10 / num
    print(f"Result: {result}")
except ValueError:
    print("That was not a number")
except ZeroDivisionError:
    print("Cannot divide by zero")
except Exception as e:
    print(f"Something went wrong: {e}")
else:
    print("Everything went fine")
finally:
    print("This always runs")
